From 013a23bf1f8fbe45ac63655e7ca115b738432595 Mon Sep 17 00:00:00 2001
From: Olivier Brunel <jjk@jjacky.com>
Date: Tue, 15 Oct 2013 15:06:26 +0200
Subject: [PATCH] tooltip: Fix possible wrong placement

When showing a tooltip on the edge of a monitor, the tooltip could be wrongly
placed and be shown going from one monitor to the next.

This happened because the current_window wasn't set visible, and when it wasn't
the returned allocated size would be 1, hence wrong calculations.

Signed-off-by: Olivier Brunel <jjk@jjacky.com>
---
 gtk/gtktooltip.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gtk/gtktooltip.c b/gtk/gtktooltip.c
index 7892ef0..12e49ae 100644
--- a/gtk/gtktooltip.c
+++ b/gtk/gtktooltip.c
@@ -1096,6 +1096,7 @@ gtk_tooltip_position (GtkTooltip *tooltip,
 #define MAX_DISTANCE 32
 
   gtk_widget_realize (GTK_WIDGET (tooltip->current_window));
+  gtk_widget_set_visible (GTK_WIDGET (tooltip->current_window), TRUE);
 
   tooltip->tooltip_widget = new_tooltip_widget;
 
-- 
1.8.4

